repository/src/main/java/com/epam/esm/repository/PaginationContext.java

```diff
@@ -0,0 +1,53 @@
+package com.epam.esm.repository;
+
+import com.epam.esm.exception.InvalidPageContextException;
+import org.springframework.stereotype.Component;
+
+import static com.epam.esm.ErrorType.INVALID_PAGE_NUMBER;
+import static com.epam.esm.ErrorType.INVALID_PAGE_SIZE;
+
+@Component
+public class PaginationContext {
+
+    private static final int MIN_PAGE = 1;
+    private static final int MIN_PAGE_SIZE = 0;
+    private static final int MAX_PAGE_SIZE = 50;
+    private static final int DEFAULT_PAGE = 1;
+    private static final int DEFAULT_PAGE_SIZE = 10;
+
+    private int page;
+    private int pageSize;
+
+    private PaginationContext(){}
+
+    public PaginationContext of(Integer page, Integer pageSize) {
+        if (page == null) {
+            page = DEFAULT_PAGE;
+        }
+
+        if (pageSize == null) {
+            pageSize = DEFAULT_PAGE_SIZE;
+        }
+
+        this.page = page;
+        this.pageSize = pageSize;
+
+        if (page < MIN_PAGE) {
+            throw new InvalidPageContextException(INVALID_PAGE_NUMBER, page);
+        }
+
+        if (pageSize < MIN_PAGE_SIZE || pageSize > MAX_PAGE_SIZE) {
+            throw new InvalidPageContextException(INVALID_PAGE_SIZE, pageSize);
```

Comment:
I think that the validation logic can be put in a separate place

Reply:
fixed